fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/NetworkChecker.kt

```diff
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.shopping.middleware
+
+import android.content.Context
+import android.net.ConnectivityManager
+import androidx.core.content.getSystemService
+import org.mozilla.fenix.ext.isOnline
+
+/**
+ * Checks if the device is connected to the internet.
+ */
+interface NetworkChecker {
+
+    /**
+     * @return true if the device is connected to the internet, false otherwise.
+     */
+    fun isConnected(): Boolean
+}
+
+/**
+ * @see [NetworkChecker].
+ */
+class NetworkCheckerImpl(private val context: Context) : NetworkChecker {
```

Comment:
**Just an observation** We are using the `Impl` suffix for the class implementation of the interface. We currently have a mix usage of `Default` prefix and `Impl` suffix. I think this is okay for new code especially in their own component and package, however, it does introduce some disparity in terms of how to name things in the codebase. This is just something to think about in terms of how we want to keep maintaining our codebase since having a consistent pattern throughout the app does help.

Reply:
That's a good observation and I agree about having consistent naming pattern across the project. I created this [bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1854065) to update the naming in the shopping package soon.